### PR TITLE
Azure provider support

### PR DESCRIPTION
**Pull request by sebbrandt87**
- addons api v1beta1
- azuredisk-csi-driver
- azurediskprovisioner
- all other usable addons activated

### CODE REVIEW

File: templates/azuredisk-csi-driver.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+apiVersion: kubeaddons.mesosphere.io/v1beta1
+kind: ClusterAddon
+metadata:
+  name: azuredisk-csi-driver
+  labels:
+    kubeaddons.mesosphere.io/name: azuredisk-csi-driver
+    kubeaddons.mesosphere.io/provides: csi-driver
+  annotations:
```
Comment (shaneutt) [inline templates/azuredisk-csi-driver.yaml]: We have a temporarily required annotation for Addons in this repository to be compatible with Catalog:

```suggestion
  annotations:
    catalog.kubeaddons.mesosphere.io/addon-revision: "0.4.0-1"
```

File: templates/azurediskprovisioner.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+apiVersion: kubeaddons.mesosphere.io/v1beta1
+kind: ClusterAddon
+metadata:
+  name: azurediskprovisioner
+  labels:
+    kubeaddons.mesosphere.io/name: azurediskprovisioner
+    kubeaddons.mesosphere.io/provides: storageclass
+  annotations:
```
Comment (shaneutt) [inline templates/azurediskprovisioner.yaml]:
```suggestion
  annotations:
    catalog.kubeaddons.mesosphere.io/addon-revision: "1.0.0-1"
```

File: templates/localvolumeprovisioner.yaml [hunk 3]
```diff
@@ -32,8 +32,20 @@ spec:
...
-    - name: docker
-      enabled: true
+    - name: azure
+      enabled: false
+      values: |
+        # Multiple storage classes can be defined here. This allows to, e.g.,
+        # distinguish between different disk types.
+        # For each entry a storage class '$name' and
+        # a host folder '/mnt/$dirName' will be created. Volumes mounted to this
+        # folder are made available in the storage class.
```
Comment (shaneutt) [inline templates/localvolumeprovisioner.yaml]: :+1: comments++

**Review by shaneutt — CHANGES_REQUESTED** [review]
Looks good, just a couple missing annotations.

File: templates/azuredisk-csi-driver.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    appversion.kubeaddons.mesosphere.io/azuredisk-csi-driver: "0.4.0"
+    values.chart.helm.kubeaddons.mesosphere.io/azuredisk-csi-driver: "https://raw.githubusercontent.com/mesosphere/charts/94333f633ccb198574f55bf0618ea182c727657a/stable/azuredisk-csi-driver/values.yaml"
+spec:
+  kubernetes:
+    minSupportedVersion: v1.15.0
+  namespace: kube-system
+  cloudProvider:
+    - name: aws
+      enabled: false
```
Comment (s12chung) [inline templates/azuredisk-csi-driver.yaml]: think you can remove this, as `aws` will never use this?

File: templates/azurediskprovisioner.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    appversion.kubeaddons.mesosphere.io/azurediskprovisioner: "1.0"
+    values.chart.helm.kubeaddons.mesosphere.io/azurediskprovisioner: "https://raw.githubusercontent.com/mesosphere/charts/efef983665c5bb1bef0901c4db26e90a32018cc8/stable/azurediskprovisioner/values.yaml"
+spec:
+  kubernetes:
+    minSupportedVersion: v1.15.0
+  cloudProvider:
+    - name: aws
+      enabled: false
```
Comment (s12chung) [inline templates/azurediskprovisioner.yaml]: think you can remove this too.

File: templates/azuredisk-csi-driver.yaml (hunk 4 above)
Comment (sebbrandt87) [reply templates/azuredisk-csi-driver.yaml]: I want to be as explicit as possible ;)

File: templates/azurediskprovisioner.yaml (hunk 5 above)
Comment (sebbrandt87) [reply templates/azurediskprovisioner.yaml]: I want to be as explicit as possible ;)

**Review by s12chung — CHANGES_REQUESTED** [review]
double checked, so there's 3 states for providers when calling `konvoy init`

1. not showing in `cluster.yaml`, when there is no `cloudProvider` entry
2. showing in `cluster.yaml`, but not enabled, when there is a `cloudProvider` entry, but enabled is not there or `false`
3. `enabled` in `cluster.yaml`,  when there is a `cloudProvider` entry and `enabled` is `true`

so it'd be better for those cloudProvider entries to be deleted.

https://github.com/mesosphere/kubeaddons/blob/74de967242d8d5e9a1e2103c335df4a3de0f8c2b/pkg/kubeaddons/v1beta1/utils.go#L134-L148

**Review by shaneutt — DISMISSED** [review]

**Review by shaneutt — APPROVED** [review]

**Review by s12chung — APPROVED** [review]

**Comment by hectorj2f** [comment]
@s12chung  We shouldn't have merged this PR before getting the approval to other related PRs for Azure :/. Anyway let's keep it IN for now :).

**Comment by s12chung** [comment]
Ah... my bad, saw it was ready...

**Pull request merged**